config/crd/bases/infrastructure.cluster.x-k8s.io_awsmanagedmachinepools.yaml

```diff
@@ -149,6 +153,12 @@ spec:
               failureReason:
                 description: "FailureReason will be set in the event that there is a terminal problem reconciling the MachinePool and will contain a succinct value suitable for machine interpretation. \n This field should not be set for transitive errors that a controller faces that are expected to be fixed automatically over time (like service outages), but instead indicate that something is fundamentally wrong with the Machine's spec or the configuration of the controller, and that manual intervention is required. Examples of terminal errors would be invalid combinations of settings in the spec, values that are unsupported by the controller, or the responsible controller itself being critically misconfigured. \n Any transient errors that occur during the reconciliation of MachinePools can be added as events to the MachinePool object and/or logged in the controller's output."
                 type: string
+              launchTemplateID:
```

Comment:
I'm not really happy with this being on `.status`, because we can't "take ownership" of an existing nodegroup with a new `AWSManagedMachinePool`, even if it was created with CAPA. On the other hand, putting it on `.spec` would imply you can pass your own `launchTemplateID`, but this isn't strictly possible because we synthesize them out of the `spec` + `EKSConfig`.

Reply:
It would be good to support the "take ownership" scenario. 

Would it be possible to do the following:

- if a launchtemplateid exists then explicitly use that. This could mean that a) its an 'unmanaged' launch template that has been created outside of CAPA and it doesn't have the owned tag or b) it was previously created and peristsed here and is owned (i.e. via tags)?
- if a launchtemplateid doesn't exist then create a new launch template by using `spec` + `EKSConfig` and then store the resulting launchtemplateid in the spec so when reconcile next runs then b) above applies?

Reply:
@michaelbeaumont - any thoughts on ^^^^?

Reply:
yeah that's definitely possible

Reply:
done